### PR TITLE
Add focusable prop to radio button

### PR DESCRIPTION
**Pull request by naamagr**
### 🔦 Summary
This enables setting a radio button as unfocusable. It is required for team-managment page where the radio buttons are for display only and not functional (selection is made by another component).

### ✅ Checklist
- [X ] 👨‍💻 API change is approved by the UI Infra Developers @shlomitc 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Other (explain)
- 🔬 Change is tested
  - [x] Component
  - [ ] Visual test

* There is no storybook page for single radio buttons (only radio groups) so I wasn't sure where this should be documented.

### CODE REVIEW

File: src/RadioGroup/RadioButton/RadioButton.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ class RadioButton extends WixComponent {
     style: PropTypes.object,
     type: PropTypes.oneOf(['default', 'button']),
     lineHeight: PropTypes.string,
+    focusable: PropTypes.bool,
```
Comment (shlomitc) [inline src/RadioGroup/RadioButton/RadioButton.js]: why not just pass `tabIndex`? this makes a more flexible and consistent API across the system

File: src/RadioGroup/RadioButton/RadioButton.driver.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ const radioButtonDriverFactory = ({ element }) => {
     isChecked: () => radioButton.checked,
     isDisabled: () => radioButton.disabled,
     getLabel: () => label.textContent,
+    getButtonTabIndex: () => element.getAttribute('tabIndex'),
```
Comment (shlomitc) [inline src/RadioGroup/RadioButton/RadioButton.driver.js]: calling it `getTabIndex` should be enough

File: src/RadioGroup/RadioButton/RadioButton.spec.js [hunk 3]
```diff
@@ -59,6 +59,24 @@ describe('RadioButton', () => {
     expect(onChange).not.toBeCalledWith(value);
   });
 
+  it('should be focusable by default', () => {
```
Comment (shlomitc) [inline src/RadioGroup/RadioButton/RadioButton.spec.js]: as you will change prop to be just `tabIndex`, I think it's better for the tests to not mentions "focusable"

**Comment by argshook** [comment]
> radio buttons are for display only and not functional 

this essentially says that radio buttons are disabled. For that only `disabled` prop should be used without any additional things.

if when `disabled={true}` RadioButton is still focusable, then that should be fixed. No new prop needed imo

**Comment by shlomitc** [comment]
The case Naama is trying to solve is a RadioButton inside a Thumbnail.
It is a valid case for the Thumbnail to be focusable and tab-able, while the RadioButton isn't and will be set as "checked" in a controlled way.

**Comment by shlomitc** [comment]
@naamagr - please add a screenshot showing your use-case

**Comment by naamagr** [comment]
image
@argshook 
@shlomitc I've pushed a fix. please re-review :)

**Pull request merged**